prov/gni/src/gnix_cm_nic.c

```diff
@@ -190,24 +190,28 @@ static int __process_datagram(struct gnix_datagram *dgram,
 /*******************************************************************************
  * Internal API functions
  ******************************************************************************/
-/*
- * generate a cdm_id, use the 16 LSB of base_id from domain
- * with 16 MSBs being obtained from atomic increment of
- * a local variable.
- */
 
-int _gnix_get_new_cdm_id(struct gnix_fid_domain *domain, uint32_t *id)
+int _gnix_cm_nic_create_cdm_id(struct gnix_fid_domain *domain, uint32_t seed,
+			       uint32_t *id)
 {
 	uint32_t cdm_id;
-	int v;
 
-	v = atomic_inc(&gnix_id_counter);
 	cdm_id = (domain->cdm_id_seed & 0x0000FFFF) |
```

Comment:
The reason for the masking is owing to the fact that currently the ```domain->cdm_id_seed``` value is obtained from ```getpid()```.   Unless there are more than 2^16 processes running concurrently on the node (really with the OS), the idea would be that this 16 LSBs should be unique.  The upper 16 bits are then used to create unique cdm_id's within the process.  In reality we could probably get away with fewer bits.